app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -239,11 +239,19 @@ class HomeFragment : Fragment() {
             }
         }
 
-        view.menuButton.setOnClickListener {
-            homeMenu?.menuBuilder?.build(requireContext())?.show(
-                anchor = it,
-                orientation = BrowserMenu.Orientation.DOWN
-            )
+        with(view.menuButton) {
+            var menu: PopupWindow? = null
+            setOnClickListener {
+                if (menu == null) {
+                    menu = homeMenu?.menuBuilder?.build(requireContext())?.show(
+                        anchor = it,
+                        orientation = BrowserMenu.Orientation.DOWN,
+                        onDismiss = { menu = null }
+                    )
+                } else {
+                    menu?.dismiss()
+                }
+            }
```

Comment:
Is this else branch necessary? Can't we just choose to only inflate if menu is null?

Reply:
Thanks Sawyer!
Indeed, the commit title and description should be improved to better describe the issue and the solution.
The `else` allows us to close an already opened menu if the 3-dot menu button is pressed again. This is the same behavior as for Chrome and basically the same recent implementation for the browser toolbar - https://github.com/mozilla-mobile/android-components/pull/4594

Reply:
Ah, great. I guess it makes sense for TalkBack enabled devices too :)

Thanks!